src/tests/ftest/osa/osa_offline_reintegration.py

```diff
@@ -0,0 +1,186 @@
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][3])
+
+    def get_pool_version(self):
+        """Get the pool version
+           Args: None
+           Returns : pool_version (int)
+        """
+        out = []
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][4])
+
+    def write_single_object(self):
+        """Write some data to the existing pool.
+           Args: None
+           Returns : None
+        """
+        self.pool.connect(2)
+        csum = self.params.get("enable_checksum", '/run/container/*')
+        container = DaosContainer(self.context)
+        input_param = container.cont_input_values
+        input_param.enable_chksum = csum
+        container.create(poh=self.pool.pool.handle,
+                         con_prop=input_param)
+        container.open()
+        obj = DaosObj(self.context, container)
+        obj.create(objcls=1)
+        obj.open()
+        ioreq = IORequest(self.context,
+                          container,
+                          obj, objtype=4)
+        self.log.info("Writing the Single Dataset")
+        record_index = 0
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0])
+                          * self.record_length[record_index])
+                d_key_value = "dkey {0}".format(dkey)
+                c_dkey = ctypes.create_string_buffer(d_key_value)
+                a_key_value = "akey {0}".format(akey)
+                c_akey = ctypes.create_string_buffer(a_key_value)
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+                ioreq.single_insert(c_dkey, c_akey, c_value, c_size)
+                record_index = record_index + 1
+                if record_index == len(self.record_length):
+                    record_index = 0
+
+    def run_offline_reintegration_test(self, num_pool, data=False):
+        """Run the offline reintegration without data.
+           Args: num_pool (int) : Total pools to create
+                                  for testing purpose.
+                 data (bool) : False (Pool has no data)
+                               True (Create some data in pool)
+           Returns : None
+        """
+        # Create a pool
+        pool = {}
+        pool_uuid = []
+        target_list = []
+        exclude_servers = len(self.hostlist_servers) - 1
+        nvme_size = 54000000000
+        scm_size = 6000000000
+
+        # Exclude target : random two targets
+        n = random.randint(1, 7)
+        target_list.append(n)
+        target_list.append(n+1)
+        t_string = "{},{}".format(target_list[0], target_list[1])
+
+        # Exclude rank : two ranks other than rank 0.
+        n = random.randint(1, exclude_servers)
```

Comment:
I think you can just do this
```suggestion
        rank = random.randint(1, exclude_servers)
```

Reply:
Yes... Initially, I was planning to have a list... A carry out... Will fix it...